torch_xla/csrc/nll_loss.cpp

```diff
@@ -103,7 +103,9 @@ xla::XlaOp BuildNllLoss(const xla::XlaOp& logits, const xla::XlaOp& labels,
       ValidLabelsCount(labels, ignore_index), logits_shape.element_type());
   xla::XlaComputation add_func =
       XlaHelpers::CreateAddComputation(logits_shape.element_type());
-  return xla::ReduceAll(mul, zero, add_func) / batch;
+  return xla::Select(xla::Ne(batch, zero),
```

Comment:
Since both branches of a Select are executed, it is better to:
```
batch = Select(Ne(batch, 0), batch, 1.0)
```

And then unconditionally divide.